src/main/java/org/fcrepo/audit/InternalAuditor.java

```diff
@@ -0,0 +1,261 @@
+    private static final String PREMIS_EVENT = "premis:Event";
+    private static final String PROV_EVENT = PROV + "InstantaneousEvent";
+
+    private static final String CONTENT_MOD = AUDIT + "contentModification";
+    private static final String CONTENT_REM = AUDIT + "contentRemoval";
+    private static final String METADATA_MOD = AUDIT + "metadataModification";
+
+    private static final String CONTENT_ADD = EVENT_TYPE + "ing";
+    private static final String OBJECT_ADD = EVENT_TYPE + "cre";
+    private static final String OBJECT_REM = EVENT_TYPE + "del";
+
+    private static final String PREMIS_TIME = "premis:hasEventDateTime";
+    private static final String PREMIS_OBJ = "premis:hasEventRelatedObject";
+    private static final String PREMIS_AGENT = "premis:hasEventRelatedAgent";
+    private static final String PREMIS_TYPE = "premis:hasEventType";
+
+    private static final String RDF_TYPE = "rdf:type";
+
+    private static final String HAS_CONTENT = REPOSITORY + "hasContent";
+    private static final String LAST_MODIFIED = REPOSITORY + "lastModified";
+    private static final String LAST_MODIFIED_BY = REPOSITORY + "lastModifiedBy";
+    private static final String NODE_ADDED = REPOSITORY + "NODE_ADDED";
+    private static final String NODE_REMOVED = REPOSITORY + "NODE_REMOVED";
+    private static final String PROPERTY_CHANGED = REPOSITORY + "PROPERTY_CHANGED";
+
+    /**
+     * Creates a node for the audit event under the configured container.
+     *
+     * @param event
+     * @throws RepositoryException
+     */
+    public void createAuditNode(final FedoraEvent event) throws RepositoryException, IOException {
+        final String uuid = pidMinter.mintPid();
+        final String userData = event.getUserData();
+        final ObjectMapper mapper = new ObjectMapper();
+        final JsonNode json = mapper.readTree(userData);
+        final String userAgent = json.get("userAgent").asText();
+        final String baseURL = json.get("baseURL").asText();
+        final String path = event.getPath();
+        final Long timestamp =  event.getDate();
+        final DateFormat df = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
+        df.setTimeZone(TimeZone.getTimeZone("UTC"));
+        final String eventDate = df.format(new Date(timestamp));
+        final String userID = event.getUserID();
+        final String eventType = getEventURIs(event.getTypes());
+        final String properties = Joiner.on(',').join(event.getProperties());
+        final String auditEventType = getAuditEventType(eventType, properties);
+        final Node auditNode = containerService.findOrCreate(session, "/audit/" + uuid).getNode();
+
+        LOGGER.debug("Audit node {} created for event.", uuid);
+
+        auditNode.addMixin("fedora:Resource");
+        auditNode.setProperty(RDF_TYPE, INTERNAL_EVENT);
+        auditNode.setProperty(RDF_TYPE, PREMIS_EVENT);
+        auditNode.setProperty(RDF_TYPE, PROV_EVENT);
```

Comment:
I'm only seeing the prov:InstantaneousEvent in the types list -- so I think you should use the setProperty() method that takes an array of values instead of a single value to get them all added.